fvm/environment/random_generator.go

```diff
@@ -0,0 +1,142 @@
+package environment
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/storage/state"
+	"github.com/onflow/flow-go/fvm/tracing"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/trace"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/state/protocol/prg"
+)
+
+type RandomGenerator interface {
+	// UnsafeRandom returns a random uint64
+	// Todo: rename to Random() once Cadence interface is updated
```

Comment:
I did think of the same idea, but then I didn't implement it because:
- the naming on the Cadence side is under discussion, so I didn't want to assume it will change.
- the PR updating Cadence version in flow-go could update the names at the same time (it will also force us to do it vs do it later in the future and risking to forget about it)